src/views/Prediction/Prediction.js

```diff
@@ -0,0 +1,147 @@
+import {splitsRequested} from '../../actions/SplitActions';
+import {mapJobs} from '../../util/unNormalize';
+import {CardText} from 'react-md/lib/Cards/index';
+import ModelSelector from '../../components/prediction/ModelSelector';
+import {REGRESSION} from '../../reference';
+import {Card} from 'react-md';
+
+class Prediction extends Component {
+  onChangeLog(logId) {
+    const log = this.props.logs.byId[logId];
+    const pLength = log.properties.maxEventsInLog;
+    this.props.onLogChange(logId, pLength);
+    this.props.onChangeJRun(logId);
+  }
+
+  onModelChange({method}, modelId) {
+    if (method === REGRESSION) {
+      this.props.onRegModelChange(modelId);
+      const classId = this.props.classModelId;
+      this.props.onModelChange(modelId, classId);
+    } else {
+      const regId = this.props.regModelId;
+      this.props.onModelChange(regId, modelId);
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.models.length === 0) {
+      this.props.onRequestModels();
+      this.props.onRequestSplitList();
+      this.props.onRequestLogList();
+      this.props.onRequestJobs();
+    }
+  }
+
+  requestJobsRun() {
+    this.props.onRequestJobs();
+  }
+
+  onReset() {
+    window.location.reload();
+  }
+
+  filterJobRun() {
+    return this.props.jobsrun.filter((job) => (job.config.log_id === this.props.logId) &&
+      ((job.config.model_id === this.props.regModelId) ||
+        (job.config.model_id === this.props.classModelId)));
+  }
+
+  Submit() {
+    const logId = this.props.logId;
+    const regId = this.props.regModelId;
+    const classId = this.props.classModelId;
+    const payload = logId + '&' + regId + '&' + classId;
+    this.props.onSubmitPrediction(payload);
+    this.props.onRequestJobs();
+  }
+
+  render() {
+    // Only unique splits for selector
+    const filteredJobsRun = this.filterJobRun();
+    const regModelsLabel = modelsToString(this.props.regressionModels);
+    const clasModelsLabel = modelsToString(this.props.classificationModels);
+
+    return (
+      <div className="md-grid">
+        <div className="md-cell md-cell--12">
+          <LogSelector logs={this.props.logs} fetchState={this.props.logfetchState}
+                       logChange={this.onChangeLog.bind(this)} logId={this.props.logId}/>
+        </div>
+        <div className="md-cell md-cell--12">
+          <ModelSelector modelChange={this.onModelChange.bind(this)} onSubmit={this.Submit.bind(this)}
+                         onReset={this.onReset} classModelsLabel={clasModelsLabel} regModelsLabel={regModelsLabel}
+                         classModelId={this.props.classModelId} regModelId={this.props.regModelId}/>
+        </div>
+        <div className="md-cell md-cell--12">
+          <Card>
+            <CardText>
+              <ResultTable jobs={filteredJobsRun} onRequestJobs={this.requestJobsRun.bind(this)}/>
+            </CardText>
+          </Card></div>
+      </div>
+    );
+  }
+}
+
+Prediction.propTypes = {
```

Comment:
there looks to be way too many props and maps and dispatches here. go check if all are really needed